python3Packages.invocations: init at 2.6.0

samuela opened the pull request:
###### Description of changes
Add `invocations` and its dependency `releases`.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

fabaff commented on pkgs/development/python-modules/invocations/default.nix:
```diff
@@ -0,0 +1,41 @@
+  propagatedBuildInputs = [ blessings invoke releases semantic-version tabulate twine ];
+
+  # There's an error loading the test suite: https://gist.github.com/samuela/ade41eb30b3029b5f4b706d7c012b2bd.
+  doCheck = false;
```
```suggestion
  doCheck = false;

  pythonImportsCheck = [
    "invocations"
  ];
```

samuela replied:
oh good catch! I forgot the `pythonImportsCheck`s.

samuela replied:
Addressed in https://github.com/NixOS/nixpkgs/commit/82761fb688373e2ca6f08b1f4e718a4f5859eaba and https://github.com/NixOS/nixpkgs/commit/5e7bced95e9e0114653435528c1cdb8e03d48491.

fabaff commented on pkgs/development/python-modules/releases/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+  propagatedBuildInputs = [ semantic-version sphinx ];
+
+  # No tests...
```
The repos contains a folder [tests](https://github.com/bitprophet/releases/tree/master/tests). IT doesn't seem to be a test suite in the classical sense.

```suggestion
  # No tests
  doCheck = false;

  pythonImportsCheck = [
    "releases"
  ];
```

samuela replied:
Addressed in https://github.com/NixOS/nixpkgs/commit/82761fb688373e2ca6f08b1f4e718a4f5859eaba and https://github.com/NixOS/nixpkgs/commit/5e7bced95e9e0114653435528c1cdb8e03d48491.

SuperSandro2000 commented on pkgs/development/python-modules/releases/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+  propagatedBuildInputs = [ semantic-version sphinx ];
+
+  # No tests...
```
Thats not true https://github.com/bitprophet/releases/tree/master/tests

samuela replied:
You're right, my mistake! There's no clear way to run them however. There's a bug report for this upstream here: https://github.com/bitprophet/releases/issues/95.

SuperSandro2000 commented on pkgs/development/python-modules/invocations/default.nix:
```diff
@@ -0,0 +1,41 @@
+  propagatedBuildInputs = [ blessings invoke releases semantic-version tabulate twine ];
+
+  # There's an error loading the test suite: https://gist.github.com/samuela/ade41eb30b3029b5f4b706d7c012b2bd.
+  doCheck = false;
```
Can you try an older pytest version?

samuela replied:
I wasn't aware that was a possibility? How do I run with an older pytest version?

samuela replied:
I'm only seeing one version in the current tree: https://github.com/nixos/nixpkgs/tree/master/pkgs/development/python-modules/pytest.

SuperSandro2000 replied:
Well, no longer since https://github.com/NixOS/nixpkgs/pull/158848/commits which dropped some older pytests...

samuela commented:
Re the test suite for `releases`: Yeah, it appears that they do have a `tests/` directory, but I'm not able to get it to run the tests in any conventional way. There's a bug report for this upstream here: https://github.com/bitprophet/releases/issues/95.

I opened an issue for the invocations test suite here: https://github.com/pyinvoke/invocations/issues/29. It appears to be bc the pytest version in nixpkgs is too recent for invocations, but they also don't specify any version constraints in the dev requirements here: https://github.com/pyinvoke/invocations/blob/main/dev-requirements.txt. And I wasn't able to find any other versions of pytest in nixpkgs to try running on.

For posterity's sake here's my attempt at getting the invocations test suite to run:
```nix
  checkInputs = [ invoke mock pytest pytest-relaxed ];
  # See https://github.com/pyinvoke/invocations/issues/25 and https://github.com/pyinvoke/invocations/issues/29.
  checkPhase = ''
    inv test
    inv integration
  '';
```

samuela committed:
python3Packages.releases: init at 1.6.3

samuela committed:
python3Packages.invocations: init at 2.6.0

samuela commented:
Addressed comments in https://github.com/NixOS/nixpkgs/pull/165911/commits/82761fb688373e2ca6f08b1f4e718a4f5859eaba and https://github.com/NixOS/nixpkgs/pull/165911/commits/5e7bced95e9e0114653435528c1cdb8e03d48491.

samuela commented:
Ok, I believe I've addressed all the comments now. I'll go ahead and merge tomorrow unless anyone objects.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 165911` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>python310Packages.invocations</li>
<li>python310Packages.releases</li>
<li>python39Packages.invocations</li>
<li>python39Packages.releases</li>
</ul>
</details>
